apps/src/code-studio/pd/foorm/editor/components/FoormEntityLoadButtons.jsx

```diff
@@ -1,7 +1,11 @@
 import React from 'react';
 import {connect} from 'react-redux';
 import PropTypes from 'prop-types';
-import {Button, DropdownButton, MenuItem} from 'react-bootstrap';
+import {Button} from 'react-bootstrap';
+import loadable from '@cdo/apps/util/loadable';
+const VirtualizedSelect = loadable(() =>
+  import('@cdo/apps/templates/VirtualizedSelect')
```

Comment:
nice touch!

Reply:
that seemed to be the pattern for other uses of this library so i went with it, but thinking about it a little more, webpack should take care of this kind of thing! i'll leave it as is for a blanket refactor later

Reply:
whoa -- can you share any info on how webpack takes care of this? my knowledge of webpack is a few versions out of date.

Reply:
the intent of the loadable thing is to prevent that file from being included in the initial load.

from my understanding of how things work:

the way we currently have webpack setup (this might not be set up the best way to support dynamic imports)
![image](https://user-images.githubusercontent.com/82416901/165807146-81bfc6e1-566d-4a3b-ab98-5b85c198a2e9.png)
the VirtualizedSelect chunk is getting pulled into the common chunk, and so the dynamic import just imports that huge common chunk and doesn't actually provide any benefit (that common chunk is already included in the js that is loaded initially)

If we set up webpack to figure out its own chunking optimization instead of using these common chunks, we'd end up with a bunch of smaller modules and also I think we wouldn't need `loadable`, we could just `import()`. The newest version of webpack recommends doing this by default (older versions were not as good at doing the chunk optimization).

Reply:
> the VirtualizedSelect chunk is getting pulled into the common chunk, and so the dynamic import just imports that huge common chunk and doesn't actually provide any benefit (that common chunk is already included in the js that is loaded initially)

ah. if something else in common is using `VirtualizedSelect` without `loadable` or `import()`, then yes, using loadable here has no performance benefit. it looks like there may only be 1 or 2 places where we are failing to use `loadable` on `VirtualizedSelect`, so one possible follow-up would be to start using loadable in the other locations (if appropriate). unfortunately, we do not have a great way to detect that a certain code change causes an increase in initial bundle download size, which is what happened when someone added the first non-loadable call to VirtualizedSelect.

> If we set up webpack to figure out its own chunking optimization instead of using these common chunks, we'd end up with a bunch of smaller modules and also I think we wouldn't need `loadable`, we could just `import()`. The newest version of webpack recommends doing this by default (older versions were not as good at doing the chunk optimization).

interesting. if you do `import()` higher up the import chain (for example, if we had a rule that all entry points could only use `import()` and not `import`), then I believe webpack will handle deciding how to split all of the transitive dependencies into different chunks, like you are describing. in that case, I would agree that we do not need loadable. this is definitely worth exploring.

however, we wouldn't want to rewrite our existing calls to `loadable` to call `import()` directly. `loadable` is just a wrapper for `import()` which makes it so that the `import()` is deferred until the time the react component is rendered, so it is still useful if we are going to use it in this way. if we stay on this path, the latest version of react has a replacement for `loadable`, which is `React.lazy`, which we should consider upgrading to: https://reactjs.org/docs/code-splitting.html#reactlazy

Reply:
oops, forgot this image:
![Screen Shot 2022-04-28 at 11 02 09 AM](https://user-images.githubusercontent.com/8001765/165818977-9e2d5d00-fb07-4e65-9b6f-212c1b9f27d6.png)

Reply:
> it looks like there may only be 1 or 2 places where we are failing to use `loadable` on `VirtualizedSelect`, so one possible follow-up would be to start using loadable in the other locations (if appropriate). unfortunately, we do not have a great way to detect that a certain code change causes an increase in initial bundle download size, which is what happened when someone added the first non-loadable call to VirtualizedSelect.

ok, actually I am not sure I understand what's going on with VirtualizedSelect and loadable. it looks like we do always wrap it with loadable and import(). I would have guessed that this means it was not in the common bundle.

Reply:
using `yarn build:dist:analyze`, VirtualizedSelect does not appear to be in our common bundle. it appears to be in a numbered module which is eligible for delay-loading:
![Screen Shot 2022-04-28 at 11 24 36 AM](https://user-images.githubusercontent.com/8001765/165821468-051b9eea-6631-4372-b598-c6e6bdabc605.png)

![Screen Shot 2022-04-28 at 11 27 24 AM](https://user-images.githubusercontent.com/8001765/165821806-90be49a4-9bcf-4ec0-b035-d85d0d1beac9.png)

sorry to go overboard on this topic... i did some investigation into webpack bundling a while back and got excited :-)

Reply:
oh, that's good to know! i just did a quick search and saw that it was included in the 0.js, which i assumed was the common bundle since it was so large and had a bunch of modules in it, but maybe the other stuff in there are all just dependencies of react-select-virtualized.